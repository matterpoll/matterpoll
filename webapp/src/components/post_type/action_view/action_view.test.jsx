import React from 'react';
import {shallow} from 'enzyme';

import ActionView from 'components/post_type/action_view/action_view';

describe('components/post_type/action_view/ActionView', () => {
    const samplePollId = 'samplepollid1';
    const baseProps = {
        post: {
            id: 'post_id',
            props: {
                poll_id: samplePollId,
            },
        },
        attachment: {
            actions: [
                {id: 'action_id1', name: 'answer1', type: 'button'},
                {id: 'action_id2', name: 'answer2', type: 'button'},
                {id: 'action_id3', name: 'answer3', type: 'button'},
            ],
        },
        votedAnswers: {
            samplepollid1: ['answer1', 'answer2'],
        },
        siteUrl: 'http://localhost:8065',
        actions: {
            fetchVotedAnswers: jest.fn(),
        },
    };

    test('should match snapshot', () => {
        const wrapper = shallow(<ActionView {...baseProps}/>);
        expect(wrapper).toMatchSnapshot();
    });
    test('should match snapshot without any actions', () => {
        const newProps = {
            ...baseProps,
            attachment: {actions: []},
        };
        const wrapper = shallow(<ActionView {...newProps}/>);
        expect(wrapper).toMatchSnapshot();
    });
    test('should match snapshot with only button actions', () => {
        const newProps = {
            ...baseProps,
            attachment: {
                actions: [
                    {id: 'action_id1', name: 'answer1', type: 'button'},
                    {id: 'action_id2', name: 'answer2', type: 'select'},
                ],
            },
        };
        const wrapper = shallow(<ActionView {...newProps}/>);
        expect(wrapper).toMatchSnapshot();
    });
    test('should match snapshot with only non-empty aciton id and name', () => {
        const newProps = {
            ...baseProps,
            attachment: {
                actions: [
                    {id: 'action_id1', name: 'answer1', type: 'button'},
                    {id: 'action_id2', name: '', type: 'button'},
                    {id: '', name: 'answer3', type: 'button'},
                    {id: 'action_id4', name: 'answer4', type: 'button'},
                ],
            },
        };
        const wrapper = shallow(<ActionView {...newProps}/>);
        expect(wrapper).toMatchSnapshot();
    });
});
