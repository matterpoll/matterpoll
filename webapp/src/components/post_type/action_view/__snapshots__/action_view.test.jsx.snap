// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/post_type/action_view/ActionView should match snapshot 1`] = `
<div
  className="attachment-actions"
>
  <Connect(ActionButton)
    action={
      Object {
        "id": "action_id1",
        "name": "answer1",
        "type": "button",
      }
    }
    key="action_id1"
    postId="post_id"
  />
  <Connect(ActionButton)
    action={
      Object {
        "id": "action_id2",
        "name": "answer2",
        "type": "button",
      }
    }
    key="action_id2"
    postId="post_id"
  />
  <Connect(ActionButton)
    action={
      Object {
        "id": "action_id3",
        "name": "answer3",
        "type": "button",
      }
    }
    key="action_id3"
    postId="post_id"
  />
</div>
`;

exports[`components/post_type/action_view/ActionView should match snapshot with only button actions 1`] = `
<div
  className="attachment-actions"
>
  <Connect(ActionButton)
    action={
      Object {
        "id": "action_id1",
        "name": "answer1",
        "type": "button",
      }
    }
    key="action_id1"
    postId="post_id"
  />
</div>
`;

exports[`components/post_type/action_view/ActionView should match snapshot with only non-empty aciton id and name 1`] = `
<div
  className="attachment-actions"
>
  <Connect(ActionButton)
    action={
      Object {
        "id": "action_id1",
        "name": "answer1",
        "type": "button",
      }
    }
    key="action_id1"
    postId="post_id"
  />
  <Connect(ActionButton)
    action={
      Object {
        "id": "action_id4",
        "name": "answer4",
        "type": "button",
      }
    }
    key="action_id4"
    postId="post_id"
  />
</div>
`;

exports[`components/post_type/action_view/ActionView should match snapshot without any actions 1`] = `""`;
